### PR TITLE
Automatically transform phone numbers to international standard

### PR DESCRIPTION
**Pull request by charludo**
### Short description

Automatically transform phone numbers to international standard.

### Proposed changes

- prepend phone number link targets with `+<country code>`
- prettyprint the internationalized phone numbers for better readability

### Side effects

- none

### Resolved issues

Fixes: #908 

__________________________________________________

[Pull Request Review Guidelines](https://digitalfabrik.github.io/integreat-cms/pull-request-review-guide.html)

### CODE REVIEW

**Comment by charludo** [comment]
(Sorry, I mis-clicked when filtering for issues, it's not Q3 yet... 🤦🏼‍♀️ By the time I realized it was already done.)

**Review by MizukiTemma — APPROVED** [review]
Nice 😸 

>(Sorry, I mis-clicked when filtering for issues, it's not Q3 yet... 🤦🏼‍♀️ By the time I realized it was already done.)

🙈 Anyway it's an improvement 😉

**Review by JoeyStk — COMMENTED** [review]
Thank you very much for this PR. It looks very good already!

While testing this, I noticed that telephone numbers such as 0123 456789 or 0123-456789 aren't caught by this PR (yet). (I found in our test system that municipalities use both cases and even a combination 0123 4567-89 :sweat_smile:). Do you think it's possible to include them too or would that be too difficult/too fuzzy given the empty space/dash between?

Secondly, although this is Javascript code I thought that we might still include a (pytest) test for this? My test idea would be to do a POST request to the page form entering a telephone number in a wrong format and then checking in a database request that the format is right in the database. Do you think that would work reliably and test what it should be testing? :thinking:

File: integreat_cms/static/src/js/tinymce-plugins/autolink_tel/plugin.js [hunk 1]
```diff
@@ -58,6 +58,33 @@
...
+        /* eslint-disable no-magic-numbers */
+        /* eslint-disable-next-line @typescript-eslint/switch-exhaustiveness-check */
+        switch (true) {
+            case phoneNumber.startsWith("00"):
+                phoneNumberCallable = `+${phoneNumber.substring(2)}`;
+                phoneNumberBody = phoneNumber.substring(4);
+                countryCode = `+${phoneNumber.substring(2, 4)}`;
+                break;
+            case phoneNumber.startsWith("0"):
+                phoneNumberCallable = `+49${phoneNumber.substring(1)}`;
```
[the remark this answers is not in the document]
Comment (charludo) [reply integreat_cms/static/src/js/tinymce-plugins/autolink_tel/plugin.js]: Yes, absolutely, good catch!

**Comment by charludo** [comment]
> While testing this, I noticed that telephone numbers such as 0123 456789 or 0123-456789 aren't caught by this PR (yet). (I found in our test system that municipalities use both cases and even a combination 0123 4567-89 😅). Do you think it's possible to include them too or would that be too difficult/too fuzzy given the empty space/dash between?

Yeah, I think this is a good idea. An OK compromise would maybe be to match all strings that 
- consist only of numbers, space, dash
- start with 0 or +
- have 6-10 digits in them

right?

Only trouble is it would for example also match `+ ---- 1 ---- 2 ----- 3 ----- 4        - 55` though I am having trouble finding a usecase for this 😂 

> Secondly, although this is Javascript code I thought that we might still include a (pytest) test for this? My test idea would be to do a POST request to the page form entering a telephone number in a wrong format and then checking in a database request that the format is right in the database. Do you think that would work reliably and test what it should be testing? 🤔

Not so easy, unfortunately. AFAIK this would not trigger the rewrite, since it doesn't happen when POSTing to the backend, only when pressing space/enter after entering such a number.

**Comment by charludo** [comment]
@JoeyStk dash is now included. It would also be an option to only allow dashes after a certain amount of numbers (e.g. if we assume all phone numbers start with either a country code +xx, 00xx, or a "Vorwahl" 0xxxx), but not sure if that wouldn't be too strict or could potentially miss some weird ways to write the numbers.

**Review by JoeyStk — CHANGES_REQUESTED** [review]
Thank you so much for the change. I double-checked with @vthejay and the most important cases are now being handled. Apparently it's no problem that space is not supported as they discourage municipalities to do use this format in the workshops already. 

I found one small last thing: When I type 0123/456789 and hit enter I get "+49 (0) 123/456789" back. @vthejay and I agreed that this isn't ideal as the "/" should not show in the final string anymore :) Do you think it's possible to remove it? :)

**Review by JoeyStk — APPROVED** [review]
:rocket:

**Pull request merged**